src/components/LeftSidebar/index.jsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from "react";
+import { NavLink } from "react-router-dom";
+import { Home, Explore, Bookmark, AccountCircle, Add } from "@mui/icons-material";
+import { v4 as uuid } from "uuid";
+import { NewPostModal } from "features";
+
+const LeftSidebar = () => {
+	const getActiveClassName = ({ isActive }) =>
+		isActive ? `active-link link` : `link`;
+
+    const [showModal, setShowModal] = useState(false);
+    
+    useEffect(() => {
+        if(showModal) {
+            document.body.style.overflow = 'hidden';
+        }
+        else {
+            document.body.style.overflow = 'auto';
+        }
+    }, [showModal]);
+
+	const sidebarLinks = [
+		{
+			id: uuid(),
+			link: "Home",
+			icon: <Home />,
+			to: "/",
+		},
+		{
+			id: uuid(),
+			link: "Explore",
+			icon: <Explore />,
+			to: "/explore",
+		},
+		{
+			id: uuid(),
+			link: "Bookmarks",
+			icon: <Bookmark />,
+			to: "/bookmarks",
+		},
+		{
+			id: uuid(),
+			link: "Profile",
+			icon: <AccountCircle />,
+			to: "/profile",
+		},
+	];
+
+	const getMappedLinks = (linkFor = "footer") =>
+		sidebarLinks.map(({ to, id, link, icon }) => (
+			<li key={id}>
+				<NavLink to={to} className={getActiveClassName}>
+					{icon} {linkFor === "sidebar" ? link : null}
+				</NavLink>
+			</li>
+		));
+
+    const handleShowModalDialog = ( modalVisibilityState) => {
+        setShowModal(modalVisibilityState);
+    }
+
+	return (
+		<>
+            { showModal ? <NewPostModal handleShowModalDialog={handleShowModalDialog} /> : null }
+			<aside className="sidebar-nav py-8 p-4 pr-8 hidden md:flex flex-col justify-start items-center h-full w-full max-w-[180px]">
+				<div className="flex fixed flex-col justify-between items-center">
+					<ul className="flex list-none flex-col gap-4 justify-center w-full">
+						{getMappedLinks("sidebar")}
+					</ul>
```

Comment:
I like how you've avoided repetitive code here by creating an array of sidebar links and mapping on that.